docs/source/developer_reference/fundamentals.rst

```diff
@@ -0,0 +1,760 @@
+to a vector document format (such as PDF or PostScript), or to a raster file
+format (such as PNG).
+
+The core object for Kiva code is the "graphics context" which represents a
+a 2D drawing surface and the current state of the drawing environment.  When
+Chaco classes need to do any drawing they will usually be supplied with an
+appropriate graphics context to render into, but very occasionally (such as
+when saving a plot out to a file) you may want to create your own context to
+draw into.
+
+If you want to write new Chaco overlays or plot renderers you should develop
+an understanding of Kiva.
+
+Enable
+------
+
+`Enable <docs.enthought.com/enable/enable>`_ provides interactivity and layout
+on top of the Kiva drawing library.  Enable is the library that handles the
+interface between Chaco plots and the OS/windowing system, as well as the
+basic hierarchical layout and layering of visible components of a plot.
+
+Enable provides two base classes that are at the root of much of the Chaco
+code:
+
+:py:class:`~enable.component.Component`
+    An object that occupies a rectangular region of the window and knows how
+    to
+    draw itself and dispatch user interactions.  The
+    :py:class:`~enable.container.Container` class is a subclass of
+    :py:class:`~enable.component.Component` that handles hierarchical layout
+    and event dispatch.
+
+    The drawing of :py:class:`~enable.component.Component` objects is split
+    across a number of layers, so that overlays, underlays, borders and
+    background can be rendered in a coherent manner.
+
+:py:class:`~enable.tools.base_tool.BaseTool`
+    An object that handles a particular type of user interaction (eg. mouse
+    events or key presses).  Each tool is a state machine and so the
+    interactions can vary depending on the state that the tool is in (eg.
+    "normal" vs. "dragging" state for a tool that handles moving
+    interactions).
+
+An understanding of Enable is important for writing new interactive tools, as
+well as for understanding and controlling how components are layered and layed
+out.
+
+Chaco Object Model
+==================
+
+The :py:class:`~.chaco.plot.Plot` class provides a fairly simple API for
+creating a plot in an application, but beneath that lies a set of classes
+that handle converting the numbers into the numpy arrays into pixels on the
+screen.  Between the :py:class:`~.chaco.array_plot_data.ArrayPlotData` and the
+:py:class:`~.chaco.plot.Plot` are a series of classes which hold state for
+various operations and transformations.
+
+Data Flow
+---------
+
+The data flow between these classes can generally be sumarised as follows:
+
+.. graphviz::
+
+    digraph dataflow {
+        rankdir=LR;
+        node [shape=plaintext];
+        "Plot Data" -> "Data sources" -> "Ranges" -> "Mappers" -> "Renderers" -> "Plot";
+        "Data sources" ->  "Renderers";
+        "Mappers" -> "Axes and Grids";
+        "Pan and Zoom" -> "Ranges";
+    }
+
```

Comment:
These graphs are not rendering for me.  I have tried installing graphviz, but still no luck.  

Down the road, maybe we ultimately will want to take the traits-futures approach (using gh actions to build docs):
https://github.com/enthought/traits-futures/blob/e63b3b0a599be595a416fa46d4ff6b20a369cd8f/.github/workflows/build-docs.yml#L19-L22